aptos-move/block-executor/src/executor.rs

```diff
@@ -714,6 +787,32 @@ where
         aggregator_v1_delta_writes
     }
 
+    /// Can be called (at most) once after transaction is committed to internally include
+    /// the resource group updates & delta outputs with the transaction outputs.
+    fn combine_groups(
+        finalized_groups: Vec<(T::Key, T::Value, Vec<(T::Tag, Arc<T::Value>)>)>,
+    ) -> ::std::result::Result<Vec<(T::Key, T::Value)>, PanicOr<IntentionalFallbackToSequential>>
+    {
+        finalized_groups
+            .into_iter()
+            .map(|(group_key, mut metadata_op, finalized_group)| {
+                let btree: BTreeMap<T::Tag, Bytes> = finalized_group
+                    .into_iter()
+                    .filter_map(|(resource_tag, arc_v)| {
```

Comment:
why is this a filter_map? do we want to be consistent to return ResourceGroupError in case of invariant violation?

Reply:
previously we don't assume unsync map / mv group map apply deletions, so we filtered them out here. However, now we can assert I think.